proposals/1984_ca-bundle-from-secret-ref/README.md

```diff
@@ -11,124 +11,261 @@ Once approved and implemented, they become historical documents.
+This function will continue to set the `TLSConfigurationValid` condition based on the validity.
 
-## Implementation
+#### `JWTAuthenticator`
+The [`validateTLS`](https://github.com/vmware-tanzu/pinniped/blob/main/internal/controller/authenticator/jwtcachefiller/jwtcachefiller.go#L248) function will be updated to read and validate certificate authority data sourced using the newly added `certificateAuthorityDataSource` field.
+This function will continue to set the `TLSConfigurationValid` condition based on the validity.
+
+
+For all the above custom resources, spanning both supervisor and concierge components, the informers that trigger their reconciliation will now watch for secrets, of type `Opaque` and `kubernetes.io/tls`, and configmaps to ensure any changes to the certificate authority data are hot loaded into respective custom resources.
+
+## Security Implications
+
+1. Pinniped will be able to read all of the data in the secrets and configmaps used to configure TLS CA bundles.
+2. For secrets of type tls, this means that Pinniped will be able to read both public and private keys in the tls secret.
+3. Pinniped will also be able to read all of the secrets and configmaps in the same namespace where Pinniped is running.
+
+## Validations and Status
+
+Most of the custom resources have a status condition of type `TLSConfigurationValid` (or something similar) which will continue to be
+used to convey validity of the certificate authority data from either `certificateAuthorityData` or `certificateAuthorityDataSource`.
+
+It is a configuration error to specify both `certificateAuthorityData` and `certificateAuthorityDataSource`, and the custom resource's status
+conditions will indicate this.
+
+When a specified secret or a configmap does not exist in the expected namepsace, or if the specified key does not exist in the secret or configmaps
+the custom resource's condition will be updated to indicate a configuration error.
+
+Status condition `TLSConfigurationValid`, or similar, only indicates whether the configuration is a parsable certificate authority data.
+Other status conditions may exist to indicate whether TLS verification itself has succeeded, which implies that the server's TLS
+certificate can be verified using the given CA bundle.
 
-Pinniped has six controllers that watch for changes in the six applicable CRs with a `certificateAuthorityData` field.
-These controllers should be enhanced with the ability to read in the named secret and perform the validations described
-above.
-In addition, the controllers should watch for changes to the named secret and reload the updated CA bundle independently
-of any changes to the containing CR.
-Controllers could watch for updates to any secret in their namespace with type `kubernetes.io/tls`, to reduce
```

Comment:
It seems worth noting that these controllers should only sync on changes to Secrets of type `Opaque` and of type `kubernetes.io/tls`. This is an important performance concern, especially for the Supervisor which is constantly doing CRUD operations on other Secret types.

Reply:
I've called this out in the new API Changes section.

Reply:
Sorry I missed that!